### PR TITLE
Feature/page transitions cherry pick

### PR DESCRIPTION
**Pull request by tyleryy**
### Overview

Page transition with sliding down lined paper graphic. Cherry picked from #72 for cleaner git history. Blocked by #62 because lined paper png is currently used instead of svg.

### CODE REVIEW

**Comment by taesungh** [comment]
For future reference, you could probably have updated #72 instead of creating an entirely new PR?

**Comment by tyleryy** [comment]
I couldn't figure out how. I think it is because for the original branch I branched off of resource-page, so I was getting duplicate commits.

File: apps/site/src/components/PageTransition/PageTransition.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+"use client";
```
Comment (taesungh) [inline apps/site/src/components/PageTransition/PageTransition.tsx:1]: Concern: Hesitant to sacrifice Server Components for this design when this becomes part of the parent layout making all children Client Components.

Comment (tyleryy) [reply apps/site/src/components/PageTransition/PageTransition.tsx:1]: working on a way to avoid wrapping everything from layout.tsx

**Comment by tyleryy** [comment]
A bit of a hack, but I delay the page route switch to let the animation complete so we don't have to wrap everything in layout.tsx and make them client components.

**Comment by tyleryy** [comment]
Also have it working for the navbar transition on `feature/page-transitions`. 

https://github.com/HackAtUCI/zothacks-site-2023/assets/66880934/39ab18c8-ed98-4f16-b01e-2150db4e382b

However, on mobile it doesn't look good. This is because I couldn't access the underlying `nav` element of the Bootstrap NavBar component because Framer Motion only works on native HTML elements.

https://github.com/HackAtUCI/zothacks-site-2023/assets/66880934/12e7227f-6ed4-4621-9e9c-4b7cef82c04f

**Comment by tyleryy** [comment]
Could use this, if necessary. https://react.dev/reference/react/useTransition

**Comment by tyleryy** [comment]
We could prevent the page transition on mobile view if we have issues also.

**Pull request closed without merging**